cfg/table.go

```diff
@@ -210,12 +210,13 @@ func InitRoutes(table table.Interface, config Config, meta toml.MetaData) error
 			table.AddRoute(route)
 		case "grafanaNet":
 
-			cfg, err := route.NewGrafanaNetConfig(routeConfig.Addr, routeConfig.ApiKey, routeConfig.SchemasFile, routeConfig.AggregationFile)
+			cfg, err := route.NewGrafanaNetConfig(routeConfig.Addr, routeConfig.ApiKey, routeConfig.SchemasFile)
```

Comment:
I reverted this change in https://github.com/grafana/carbon-relay-ng/commit/4cad65f3405056163a00399b785ba5c028a64c0c - now `NewGrafanaNetConfig` takes aggFile as a parameter, and if it's not the empty string, tries to validate it. Like you mention, in `imperatives.go` we still need to manually validate and set aggregationFile. 

My initial thought was to directly set the aggregationFile field after the config had been created, since it's an optional field, and that's how all the other optional fields are set. However, that could lead to accidentally forgetting to validate the field before we try and create a new route with the config like I originally did here, so I applied your suggestion. 

I think it's still not ideal since there are two paths for setting the aggregationFile and both need to do validation. A future improvement could be to take in option functions in the config constructor. That way setting the optional fields won't require the config instance to be created before they can be set.